File: lengthoflastword.java
public class Solution {
    public int lengthOfLastWord(String s) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int i=0;
        for(;i<s.length();i++){
            if(s.charAt(s.length()-i-1)!=' '){
                break;
            }
        }
        int t = i;
        
        for(;i<s.length();i++){
            if(s.charAt(s.length()-i-1)==' '){
                return i-t;
            }
        }
        return s.length()-t;
    }
}

File: cpp/LengthofLastWord.cpp
class Solution {
public:
    int lengthOfLastWord(const char *s) {
        int n = (int)strlen(s);
        if(n==0) {
            return 0;
        }
        int i=n-1;
        while(i>=0 && s[i]==' ') {
            i--;
        }
        int count = 0;
        while(i>=0 && s[i]!=' ') {
            count ++;
            i--;
        }
        return count;
    }
};
